### ISSUE TITLE
New TimeZone should avoid explicit conversions

### ISSUE THREAD

**Issue by jfelchner** [bbatsov/rubocop#1770]
If the intention of the `TimeZone` cop is to keep people from accidentally working with a `Time` which has a Time Zone that they are not aware of, then explicit conversions via the `utc` method should be ignored since it doesn't matter what time zone `Time.now` returns if `utc` is chained onto it.

I can definitely see a situation where the user doesn't care what the Rails time zone is set to and they want to work explicitly with `UTC` times.

Proposal:

`utc`
`localtime(offset)`

should be marked as "safe" to use after `Time.now` and any other method marked as "dangerous" by this cop.

Rationale:

`Time.now.utc` while explicitly being cast to `UTC`, is not the same as `Time.zone.now` since `Time.zone` can be something other than `UTC`.

**Comment by bbatsov** [comment]
//cc @palkan

**Comment by palkan** [comment]
Just to make sure I understand you correctly:

```ruby
# good
Time.now.utc
Time.parse(some_time).localtime("+09:00")

# still bad
Time.now.localtime
```

@bbatsov 
Do you think we should add this as 'always' or 'acceptable' style? I think the second.

**Comment by ojab** [comment]
Second that, rubocop doesn't like `Time.now.utc.strftime('%Y-%m-%d %H:%M:%S')` here.

**Comment by jfelchner** [comment]
@palkan to keep it consistent with what you currently have, I'd just mark it as `acceptable`.
